docs/design/features/host-tracing.md

```diff
@@ -1,87 +1,96 @@
+* `dotnet` or `apphost` executable - uses `hostfxr`
+* `nethost` library - no dependencies
+* `hostfxr` library - uses `hostpolicy`
+* `hostpolicy` library - no host dependencies (uses the runtime only)
+* custom host - uses `hostfxr` and possibly `nethost`
 
-## Components involved
-Host tracing spans various components:
-* `dotnet` or `apphost` executable
-* `hostfxr` library
-* `hostpolicy` library
+The trace settings must be propagated across the component boundaries, such that all components have the same external behavior.
 
-If a custom host is used it will typically replace the executables and possible even some of the libraries. The custom host can use either `hostfx` or `hostpolicy` directly. So each of these components must support tracing on its own along with the configuration. In case several of these components are used together, they must cooperate and "share" the tracing setup.
+All these components (with the exception to custom host) include the same code for tracing but act mostly independently. The fact that all the components share the same code produces the end behavior that the components all write their trace output to the same location given the same environment. The only synchronization currently implemented is that each component fully flushes all internal tracing buffers before transferring control to a different component.
 
-Currently (as of .NET Core 2.1) all these components include the same code for tracing but act completely independently. The fact that all the components share the same code produces the end behavior that the components all write their trace output to the same location (that is `stderr`). The only synchronization currently implemented is that each component fully flushes all internal tracing buffers before transferring control to a different component.
+## Trace routing
 
-## New trace routing
-The host components implement two routes for tracing:
-* Tracing to `stderr` (this is the .NET Core 2.1 behavior). This default route is activated by setting the `COREHOST_TRACE=1` environment variable.
-* Tracing to a file (new in .NET Core 3). This route is activated by setting the `COREHOST_TRACE=1` and also providing the full path to the trace file in `COREHOST_TRACEFILE=<path>` environment variable. This overrides the `stderr` route, so if the file tracing is enabled, traces will not be written to `stderr`.  Instead traces will be appended to the specified file.
+Tracing is enabled by setting `COREHOST_TRACE=1` env. variable.
+
+In .NET Core 2.1 and below the host tracing is only written to the `stderr` output of the process.
+
+Starting with .NET Core 3 tracing can be redirected and its verbosity controlled:
+
+* Redirect the trace to a file (always appends) - enable tracing via `COREHOST_TRACE=1` and also set `COREHOST_TRACEFILE=<path>` in the environment for the process. The `<path>` is resolved against current directory and the file is opened for text append write.
```

Comment:
Might also want to call out that the parent directory for the file in `COREHOST_TRACEFILE` must exist.